src/serac/numerics/mesh_utils.hpp

```diff
@@ -93,12 +100,57 @@ struct InputOptions {
   static void defineInputFileSchema(axom::inlet::Table& table);
 
   std::string relative_mesh_file_name;
+};
+
+struct GenerateInputOptions {
+  /**
+   * @brief Input file parameters for mesh generation
+   *
+   * @param[in] table Inlet's SchemaCreator that input files will be added to
+   **/
+  static void defineInputFileSchema(axom::inlet::Table& table);
+
+  /// For rectangular and cuboid meshes
+  std::vector<int>    elements;
+  std::vector<double> overall_size;
+};
+
+struct InputOptions {
+  /**
+   * @brief Input file parameters for mesh generation
+   *
+   * @param[in] table Inlet's SchemaCreator that input files will be added to
+   **/
+  static void defineInputFileSchema(axom::inlet::Table& table);
+
+  std::variant<FileInputOptions, GenerateInputOptions> extra_options;
   // Serial/parallel refinement iterations
   int ser_ref_levels;
   int par_ref_levels;
 };
 
 }  // namespace mesh
+
+/**
+ * @brief Constructs a 3D MFEM mesh of a cuboid
+ *
+ * @param[in] extra_options Cuboid Mesh Options
+ * @param[in] MPI_Comm MPI Communicator
+ * @return A shared_ptr containing the constructed mesh
+ */
+std::shared_ptr<mfem::ParMesh> buildCuboidMesh(serac::mesh::GenerateInputOptions& options,
+                                               const MPI_Comm = MPI_COMM_WORLD);
```

Comment:
Can these be named (just `comm` or something)?  I think Doxygen likes to have the parameters named in the prototype/header.

Reply:
Sure. I just copied it from the original buildCuboidMesh. I'll change them.